Release Notes for Kyma 2.10 (#939)
kyma-project/website: content/blog-posts/2023-01-12-release-notes-2.10/index.md

---
title: "Kyma 2.10"
author:
  name: "Wojciech Sołtys, RM @Kyma, and Iwona Langer, Technical Writer @Kyma"
tags:
  - release-notes 
type: release 
releaseTag: "2.10"
redirectFrom:
  - "/blog/release-notes-210"
---

The holiday season gave us a welcome break and an excellent opportunity to recharge our batteries. With this energy boost, we’ve set off into the new year, wishing you the best of luck and offering you the best of Kyma in its latest version, 2.10! Read on to find out what we have prepared for you.

<!-- overview -->

See the overview of all changes in this release:

- [Application Connectivity](#application-connectivity) – replacing `kyma-integration` Namespace with `kyma-system` Namespace
- [Eventing](#eventing) – NATS server updated to version 2.9.9
- [Observability](#observability) – Kiali and monitoring deprecation
- [Telemetry](#telemetry) – logs improvements, configurable tracing
- [Serverless](#serverless) – previous Jaeger endpoint no longer supported 
- [Service Mesh](#service-mesh) – Istio upgrade, Istio sidecar annotation deprecated, support for XFF header

## Application Connectivity

> From now Application Gateway will use the `kyma-system` Namespace instead of the `kyma-integration` Namespace. The secrets related to the Namespace are not removed automatically in the process of upgrading Kyma to version 2.10. To remove them, you must run the [migration script]( https://github.com/kyma-project/kyma/blob/main/docs/assets/2.9-2.10-OS-copy-secrets-to-system-namespace.sh) after successfully migrating from Kyma 2.9 to Kyma 2.10. 

Reviewer: ```suggestion
From now Application Gateway will use the `kyma-system` Namespace instead of the `kyma-integration` Namespace. The Secrets related to the Namespace are not removed automatically in the process of upgrading Kyma to version 2.10. To remove them, you must run the [migration script]( https://github.com/kyma-project/kyma/blob/main/docs/assets/2.9-2.10-OS-copy-secrets-to-system-namespace.sh) after successfully migrating from Kyma 2.9 to Kyma 2.10. 
```

## Eventing

> NATS server update to version 2.9.9 improves overall stability and performance, and prevents security issues. For more details, see the NATS server [Release v2.9.9]( https://github.com/nats-io/nats-server/releases/tag/v2.9.9).

Reviewer: ```suggestion
The NATS server update to version 2.9.9 improves overall stability and performance, and prevents security issues. For more details, see the NATS server [Release v2.9.9]( https://github.com/nats-io/nats-server/releases/tag/v2.9.9).
```

> The NATS server update to version 2.9.9 improves overall stability and performance, and prevents security issues. For more details, see the NATS server [Release v2.9.9]( https://github.com/nats-io/nats-server/releases/tag/v2.9.9).

Reviewer 2: Because of release blocking issues, we will have to change the release notes:

```suggestion
### NATS
Introduced a `System Account` user to allow full leverage of the NATS-cli ([Further reads](https://kyma-project.io/docs/kyma/2.10/04-operation-guides/operations/evnt-03-nats-server-system-events). 

### Metrics 
We exposed new metrics for backend latency.
Also, we have introduced consistent naming for all metrics exposed by EPP. 
Further reads: [Kyma docs](https://kyma-project.io/docs/kyma/2.10/04-operation-guides/operations/evnt-02-eventing-metrics) .
```

Author: > Because of release blocking issues, we will have to change the release notes:

Right, I'll edit the RNs.

## Service Mesh

### Istio upgraded to 1.16.1  

With release 2.10, we upgrade Istio from 1.15.3 to 1.16.1. For more details on the changes, read the official [Istio 1.16. Upgrade Notes](https://istio.io/latest/news/releases/1.16.x/announcing-1.16/upgrade-notes/).

### Istio sidecar injection annotation deprecated  

As of version 1.16 of Istio, the `sidecar.istio.io/inject` annotation was deprecated in favor of the `sidecar.istio.io/inject` label. To learn how to configure your Pods so that the Istio sidecar injection remains fully supported, read [this blog post]( https://kyma-project.io/blog/2022/12/30/deprecation-of-istio-sidecar-injection-annotation).

### Support for XFF header  

> With this version of Kyma, we introduce [Istio CR](https://kyma-project.io/docs/kyma/main/05-technical-reference/00-custom-resources/oper-01-istio/) supporting the configuration of `numTrustedProxies` Read about the [Support for XFF with Istio](https://github.com/kyma-project/website/blob/main/content/blog-posts/2023-01-11-istio-xff-support/index.md) to learn how to configure Istio CR and secure your workload on a forwarded client IP.

Reviewer: ```suggestion
With this version of Kyma, we introduce [Istio CR](https://kyma-project.io/docs/kyma/main/05-technical-reference/00-custom-resources/oper-01-istio/) supporting the configuration of `numTrustedProxies`. Read about the [Support for XFF with Istio](https://github.com/kyma-project/website/blob/main/content/blog-posts/2023-01-11-istio-xff-support/index.md) to learn how to configure Istio CR and secure your workload on a forwarded client IP.
```

> With this version of Kyma, we introduce [Istio CR](https://kyma-project.io/docs/kyma/main/05-technical-reference/00-custom-resources/oper-01-istio/) supporting the configuration of `numTrustedProxies` Read about the [Support for XFF with Istio](https://github.com/kyma-project/website/blob/main/content/blog-posts/2023-01-11-istio-xff-support/index.md) to learn how to configure Istio CR and secure your workload on a forwarded client IP.

Reviewer 3: ```suggestion
With this version of Kyma, we introduced [Istio custom resource (CR)](https://kyma-project.io/docs/kyma/main/05-technical-reference/00-custom-resources/oper-01-istio/) supporting the configuration of `numTrustedProxies`. Read about the [Support for XFF with Istio](https://github.com/kyma-project/website/blob/main/content/blog-posts/2023-01-11-istio-xff-support/index.md) to learn how to configure Istio CR and secure your workload on a forwarded client IP.
```

---
title: "Kyma 2.10"
author:
  name: "Wojciech Sołtys, RM @Kyma, and Iwona Langer, Technical Writer @Kyma"
tags:
  - release-notes 
type: release 
releaseTag: "2.10"
redirectFrom:
  - "/blog/release-notes-210"
---

The holiday season gave us a welcome break and an excellent opportunity to recharge our batteries. With this energy boost, we’ve set off into the new year, wishing you the best of luck and offering you the best of Kyma in its latest version, 2.10! Read on to find out what we have prepared for you.

<!-- overview -->

See the overview of all changes in this release:

- [Application Connectivity](#application-connectivity) – replacing `kyma-integration` Namespace with `kyma-system` Namespace
- [Eventing](#eventing) – NATS server updated to version 2.9.9

> - [Observability](#observability) – Kiali and monitoring deprecation

Reviewer 4: ```suggestion
- [Observability](#observability) – Monitoring deprecation
```
Kiali is just a reminder

## Application Connectivity
From now Application Gateway will use the `kyma-system` Namespace instead of the `kyma-integration` Namespace. The Secrets related to the Namespace are not removed automatically in the process of upgrading Kyma to version 2.10. To remove them, you must run the [migration script]( https://github.com/kyma-project/kyma/blob/release-2.10/docs/assets/2.9-2.10-OS-copy-secrets-to-system-namespace.sh) after successfully migrating from Kyma 2.9 to Kyma 2.10. 

## Eventing

### NATS
We introduced a `system account` user to allow full leverage of the NATS CLI. For more information, read about [Acquiring NATS server system account credentials](https://kyma-project.io/docs/kyma/2.10/04-operation-guides/operations/evnt-03-nats-server-system-events). 

### Metrics 
We exposed new metrics for backend latency.
We also introduced consistent naming for all metrics exposed by Eventing Publisher Proxy (EPP). 
Read [Kyma Eventing Metrics](https://kyma-project.io/docs/kyma/2.10/04-operation-guides/operations/evnt-02-eventing-metrics) for more details.

## Observability
 
### Kiali

> Kiali was deprecated with Kyma 2.8 and will be removed in Kyma 2.11. For more details, see the blog post on [Kiali deprecation](https://kyma-project.io/blog/2022/10/10/Kiali-deprecation).

Reviewer 4: Maybe put something like "Reminder" in front?

Author: @a-thaler We don't use this kind of panel in our documentation. Just NOTE, TIP or CAUTION. I suggest expressing the idea that the info was published before with the phrase: As announced in the blog post...
However, if you insist on using "Reminder", I'll change the note.

Reviewer 4: I don't insist on it. I thought we should make it more explicit that it got already deprecated and this is just another warning that the removal will come. If you think it is good enough, I'm fine :)

 
### Monitoring
Kyma 2.10 brings the update to the latest Prometheus images (version 2.40.7), which resolves the security vulnerabilities.
 
## Telemetry
 
### Logs
With Kyma 2.10, we offer solutions to the following issues:
- Improved implementation so that [unnecessary restarts of Fluent Bit DaemonSet are avoided on the LogPipeline changes](https://github.com/kyma-project/kyma/issues/15956)
- Improved [certificate handling of LogPipeline and LogParser webhook](https://github.com/kyma-project/kyma/issues/15765) so that there are no webhook downtimes caused by improper certificates 
- [Dashboard support](https://github.com/kyma-project/kyma/issues/15894) for defining LogPipelines and LogParser 
 
### Traces

> The first version of the [revamped tracing](https://kyma-project.io/docs/kyma/main/01-overview/main-areas/telemetry/telemetry-03-traces/) feature is out. Under the umbrella of the telemetry module, a new CRD TracePipeline is available. That new API will manage an OpenTelemetry Collector which supports you in integrating into OTLP-based backends. For now, Basic Authentication is supported, and further ways of authentication will be added soon.

Reviewer 4: ```suggestion
The first version of the new [configurable tracing](https://kyma-project.io/docs/kyma/main/01-overview/main-areas/telemetry/telemetry-03-traces/) feature is available. Under the umbrella of the telemetry module, a new CRD TracePipeline is available. That new API will manage an OpenTelemetry Collector which supports you integrating the trace aspects within a runtime into OTLP-based backends. For now, Basic Authentication is supported, and further ways of authentication will be added soon.
```

## Eventing

### NATS
We introduced a `system account` user to allow full leverage of the NATS CLI. For more information, read about [Acquiring NATS server system account credentials](https://kyma-project.io/docs/kyma/2.10/04-operation-guides/operations/evnt-03-nats-server-system-events). 

### Metrics 
We exposed new metrics for backend latency.
We also introduced consistent naming for all metrics exposed by Eventing Publisher Proxy (EPP). 
Read [Kyma Eventing Metrics](https://kyma-project.io/docs/kyma/2.10/04-operation-guides/operations/evnt-02-eventing-metrics) for more details.

## Observability
 
### Kiali
As announced in the blog post on [Kiali deprecation](https://kyma-project.io/blog/2022/10/10/Kiali-deprecation), Kiali was deprecated with Kyma 2.8 and will be removed in Kyma 2.11.

### Deprecation of Monitoring  
We decided to deprecate the monitoring component based on Prometheus/Grafana. Please start the transition to alternative solutions now. The actual removal will happen in half a year or later, with Kyma 2.16 at the earliest. For more information, read [this blog post](https://kyma-project.io/blog/2022/12/9/monitoring-deprecation/).
 
### Monitoring

> Kyma 2.10 brings the update to the latest Prometheus images (version 2.40.7), which resolves the security vulnerabilities.

Reviewer 5: do we want to specify image updated to resolve security vulnerabilites ? Cant we just say images have been updated ?
